### DISCUSSION TITLE
Caching doesn't work correctly with latest rc

### DISCUSSION THREAD

**Discussion by max-frai** [amannn/next-intl#576] [Q&A]
### Description

You have on landing page link to example which uses "latest" tag in package.json and everything is okay.
When I migrate app to `3.0.0-rc.6` there is a problem with correct caching headers. 
With example from landing on production build I see the following headers:

`Cache-Control: s-maxage=31536000, stale-while-revalidate`

But with rc6 header is:

`Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate`

Of course, this makes app very slow because nextjs doesn't cache ssr, etc and recalculates everything on each request.

I build a small example with rc6 which after yarn run build and yarn start on main page gives this wrong cache-control header.

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://github.com/max-frai/nextintl-cache-problem

### Reproduction description

Steps to reproduce:
1. yarn 
2. yarn run build
3. yarn run start
4. Open localhost:3000/en
5. Watch Cache-Control header

### Expected behaviour

With production build and all default values for caching it should return the following header: `Cache-Control: s-maxage=31536000, stale-while-revalidate`

**Comment by amannn** [comment]
I think you're looking for this: https://next-intl-docs-git-feat-next-13-rsc-next-intl.vercel.app/docs/getting-started/app-router#static-rendering

I'll move this to a discussion, since this appears to be a usage question.

**Reply by max-frai** [reply]
So the main idea is that for now it's not possible to make the logic that I want, when site is dyanmic: web site with hundred thousands of publications? I'm not sure that static rendering is correct case for me, what I'm looking for is cached internally ssr which rebuilds when page changes (for example, when there are new blogs on index page)

**Reply by andyngdz** [reply]
I'm also getting the same problem. Not sure if it's the next-intl's issue? It shows `private, no-cache` which slow down the site. I've also tried follow the link above which set the `generateStaticParams` and use `unstable_setRequestLocale`. However it doesn't work for the dynamic paths such as `/product/[id]`. Any suggestions? Appreciate your work @amannn

**Reply by amannn** [reply]
> dynamic paths such as `/product/[id]`

You'd need `generateStaticParams` on that route too.

There's a trick that `generateStaticParams` can return an empty array, which means: "Don't generate anything at build time, but cache dynamic requests when the app runs.".

Maybe that's helpful for your case!
